File: src/agenticflow_sdk/resources/uploads.py
"""Upload-session resource helpers."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any, Mapping

if TYPE_CHECKING:
    from ..client import AgenticFlowSDK


class UploadsResource:
    def __init__(self, client: "AgenticFlowSDK") -> None:
        self._client = client

    def input_create(
        self,
        payload: Mapping[str, Any],
        *,
        dry_run: bool = False,
    ) -> dict[str, Any]:
        operation = self._client.resolve_operation("uploads.input_create")
        return self._client.call(
            operation,
            json_body=dict(payload),
            dry_run=dry_run,
        )

    def input_status(
        self,
        *,
        session_id: str,
        dry_run: bool = False,
    ) -> dict[str, Any]:
        operation = self._client.resolve_operation("uploads.input_status")
        return self._client.call(
            operation,
            path_params={"session_id": session_id},
            dry_run=dry_run,
        )

File: packages/sdk/src/resources/uploads.ts
/**
 * Upload-session resource helpers.
 */

import type { AgenticFlowSDK } from "../core.js";

export class UploadsResource {
  constructor(private client: AgenticFlowSDK) { }

  async inputCreate(payload: Record<string, unknown>): Promise<unknown> {
    return (await this.client.post("/v1/uploads/inputs/anonymous", { json: payload })).data;
  }

  async inputStatus(sessionId: string): Promise<unknown> {
    return (await this.client.get(`/v1/uploads/sessions/${sessionId}/anonymous`)).data;
  }
}
